### ISSUE TITLE
Can I use Pideck with a Rane 62 as a soundcard?

### ISSUE THREAD

**Issue by CircuitCirkus** [pideck/pideck#54]

**Comment by danielhjames** [comment]
Potentially, it would depend on the USB audio interface(s) being class compliant. If you have the mixer already, please plug the USB into any GNU/Linux computer and issue the commands:

arecord -l
aplay -l

The output from these commands should tell us what kind of USB devices are involved, if they are recognized by ALSA.

**Comment by MrShigura** [comment]
Just wondering if this went anywhere, if not I'm curious to try the same thing and would be happy to report my results once I get to the point to test this....

**Comment by danielhjames** [comment]
Hi @MrShigura , if you have a Rane 62 please do let us know the output of the commands in my reply above.

**Comment by MrShigura** [comment]
@danielhjames Will do, might take me a bit as I'm just stumbling my way into learning pi (largely for this specific use) but I'll make sure to post it as soon as I can.

**Comment by MrShigura** [comment]
@danielhjames Have the output for you, doesn't look promising but, then again, I have no idea what I'm talking about... -_-

So much so, in fact, that I wasn't sure which command you meant, so I just went with doing both...

pi@raspberrypi:~ $ arecord -l
**** List of CAPTURE Hardware Devices ****
pi@raspberrypi:~ $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: ALSA [bcm2835 ALSA], device 0: bcm2835 ALSA [bcm2835 ALSA]
  Subdevices: 8/8
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
  Subdevice #4: subdevice #4
  Subdevice #5: subdevice #5
  Subdevice #6: subdevice #6
  Subdevice #7: subdevice #7
card 0: ALSA [bcm2835 ALSA], device 1: bcm2835 ALSA [bcm2835 IEC958/HDMI]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
pi@raspberrypi:~ $ arecord -L
null
    Discard all samples (playback) or generate zero samples (capture)
pi@raspberrypi:~ $ aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
default:CARD=ALSA
    bcm2835 ALSA, bcm2835 ALSA
    Default Audio Device
sysdefault:CARD=ALSA
    bcm2835 ALSA, bcm2835 ALSA
    Default Audio Device
dmix:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Direct sample mixing device
dmix:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Direct sample mixing device
dsnoop:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Direct sample snooping device
dsnoop:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Direct sample snooping device
hw:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Direct hardware device without any conversions
hw:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Direct hardware device without any conversions
plughw:CARD=ALSA,DEV=0
    bcm2835 ALSA, bcm2835 ALSA
    Hardware device with all software conversions
plughw:CARD=ALSA,DEV=1
    bcm2835 ALSA, bcm2835 IEC958/HDMI
    Hardware device with all software conversions

**Comment by danielhjames** [comment]
Thanks @MrShigura, this output shows no detection of capture channels, so it will need some tweaking at the very least. I believe this mixer comes with a control panel on Windows which might be required to set up the routing. The best place to ask about this is on the alsa-user mailing list: https://sourceforge.net/projects/alsa/lists/alsa-user

**Comment by obbardc** [comment]
Was this ran with the mixer plugged into the Pi, turned on and the USB
switch on the mixer set to correct position?


Can you post the output of the commands:

"sudo dmesg"
and
"sudo lsusb"



Cheers!

**Comment by MrShigura** [comment]
@obbardc Yep, it was, tbh I'm not all that surprised, Rane/Serato don't tend to have a very open source attitude toward hardware. I'll give those commands a shot though when I get a chance and let you know.

**Comment by MrShigura** [comment]
@obbardc @danielhjames Results of those commands, looks like my audio injector isn't doing so hot... ah the joys of being a pi nub...

[funwithcommands.txt](https://github.com/pideck/pideck/files/1883288/funwithcommands.txt)

**Comment by djrek** [comment]
Is there any update on this?  I have a Rane 62, that I would like to have working with pideck.  No a programmer but willing to test with my mixer.

**Comment by danielhjames** [comment]
Thanks @djrek according to https://serato.com/forum/discussion/1671984 the 72 is class compliant but the 62 is not. Writing a driver for the 62 would be non-trivial. So your best bet is to use external audio interfaces connected to the analogue inputs of the 62.
